src/sympc/protocol/falcon/falcon.py

```diff
@@ -553,3 +585,156 @@ def private_compare(x: List[MPCTensor], r: torch.Tensor) -> MPCTensor:
         beta_prime = d_val
 
         return beta_2 + beta_prime
+
+    @staticmethod
+    def wrap2(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+        """Computes wrap on the input numpy array.
+
+        Args:
+            a (np.ndarray): Input numpy array
+            b (np.ndarray): Input numpy array.
+
+        Returns:
+            result (np.ndarray): Boolean array ,True if there is a carry.
+
+        Raises:
+            ValueError: If the input values is not a numpy array.
+            ValueError: If signed numpy array are provided as input.
+
+        wrap2 calucaties the carry bit on addition of two values a+b.
+        """
+        if not isinstance(a, np.ndarray) or not isinstance(b, np.ndarray):
+            raise ValueError("Input value must be a numpy array for wrap2.")
+
+        if a.dtype.kind == "i" or b.dtype.kind == "i":
+            raise ValueError("Wrap2 works only for signed numbers.")
+
+        max_val = np.iinfo(a.dtype).max
+
+        return a > max_val - b
+
+    @staticmethod
+    def wrap3(a: np.ndarray, b: np.ndarray, c: np.ndarray) -> np.ndarray:
+        """Computes wrap on the input numpy array.
+
+        Args:
+            a (np.ndarray): Input numpy array
+            b (np.ndarray): Input numpy array.
+            c (np.ndarray): Input numpy array.
+
+        Returns:
+            result (np.ndarray): Modulo reduction of the exact wrap function.
+
+        Raises:
+            ValueError: If the input values is not a numpy array.
+            ValueError: If signed numpy array are provided as input.
+
+        wrap3 calucaties the carry bit on addition of three values a+b+c(mod 2).
+        """
+        if (
+            not isinstance(a, np.ndarray)
+            or not isinstance(b, np.ndarray)
+            or not isinstance(c, np.ndarray)
+        ):
+            raise ValueError("Input value must be a numpy array for wrap3.")
+
+        if a.dtype.kind == "i" or b.dtype.kind == "i" or c.dtype.kind == "i":
+            raise ValueError("Wrap3 works only for signed numbers.")
+
+        return Falcon.wrap2(a, b) ^ Falcon.wrap2(a + b, c)
+
+    @staticmethod
+    def wrap_preprocess(a: MPCTensor, session: Session) -> List:
```

Comment:
Could you change the annotation to List of MPCTensors? Same for docstrings.

Reply:
Done